lib/websocket_lwt.ml

```diff
@@ -233,19 +237,10 @@ let establish_server
       Connected_client.create ?read_buf ?write_buf random_string request flow ic oc in
     react client
   in
-  Conduit_lwt_unix.serve ?timeout ?stop ~ctx ~mode
-    (fun flow ic oc ->
-       (try%lwt
-         set_tcp_nodelay flow;
-         server_fun flow ic oc
-        with
-        | End_of_file ->
-          Lwt_log.info ~section "Client closed connection"
-        | Exit ->
-          Lwt_log.info ~section "Server closed connection"
-        | exn -> exception_handler exn; Lwt.return_unit
-       ) [%finally Lwt_io.close ic]
-    )
+  Conduit_lwt_unix.serve ?on_exn ?timeout ?stop ~ctx ~mode begin fun flow ic oc ->
+    set_tcp_nodelay flow;
+    server_fun flow ic oc
+  end
```

Comment:
Does `Conduit_lwt_unix.serve` close the connection automatically (when `server_fun` returns or raises)?